worker/job_completed_worker.go

```diff
@@ -37,39 +40,50 @@ const nameJobCompleted = "job_completed_worker"
 
 // JobCompletedWorker is the JobCompletedWorker struct
 type JobCompletedWorker struct {
-	Logger         zap.Logger
-	MarathonDB     *extensions.PGClient
-	Config         *viper.Viper
-	SendgridClient *extensions.SendgridClient
+	Workers *Worker
+	Logger  zap.Logger
 }
 
 // NewJobCompletedWorker gets a new JobCompletedWorker
-func NewJobCompletedWorker(config *viper.Viper, logger zap.Logger) *JobCompletedWorker {
+func NewJobCompletedWorker(workers *Worker) *JobCompletedWorker {
 	b := &JobCompletedWorker{
-		Config: config,
-		Logger: logger.With(zap.String("worker", "JobCompletedWorker")),
+		Logger:  workers.Logger.With(zap.String("worker", "JobCompletedWorker")),
+		Workers: workers,
 	}
-	b.configure()
-	log.D(logger, "Configured JobCompletedWorker successfully.")
+	b.Logger.Debug("Configured JobCompletedWorker successfully.")
 	return b
 }
 
-func (b *JobCompletedWorker) configureMarathonDatabase() {
-	var err error
-	b.MarathonDB, err = extensions.NewPGClient("db", b.Config, b.Logger)
+func (b *JobCompletedWorker) flushControlGroup(job *model.Job) {
+	hash := job.ID.String()
+	hash = fmt.Sprintf("%s-CONTROL", hash)
+	controlGroup, err := b.Workers.RedisClient.LRange(hash, 0, -1).Result()
 	checkErr(b.Logger, err)
-}
 
-func (b *JobCompletedWorker) configureSendgrid() {
-	apiKey := b.Config.GetString("sendgrid.key")
-	if apiKey != "" {
-		b.SendgridClient = extensions.NewSendgridClient(b.Config, b.Logger, apiKey)
+	folder := b.Workers.Config.GetString("s3.controlGroupFolder")
+	csvBuffer := &bytes.Buffer{}
+	csvWriter := io.Writer(csvBuffer)
+	csvWriter.Write([]byte("controlGroupUserIds\n"))
+	for _, user := range controlGroup {
+		csvWriter.Write([]byte(fmt.Sprintf("%s\n", user)))
 	}
+
+	bucket := b.Workers.Config.GetString("s3.bucket")
+	writePath := fmt.Sprintf("%s/%s/job-%s.csv", bucket, folder, job.ID.String())
```

Comment:
isn't this the same path as the csv one?

Reply:
No, this is the control group path.

Reply:
`s3.controlGroupFolder` my bad!